lib/hash_pipe/array_conversion.rb

```diff
@@ -0,0 +1,27 @@
+Array.class_eval do
+  # Creates a deep object out of an array
+  #
+  # hash = ["errors", "something", "somethingElse", "password",  ["VALIDATION_STATE_PASSWORD"]]
+  #
+  # Could be turned into an object usign this
+  #
+  # hash.nest
+  #
+  # And the end result would look like this:
+  #
+  # "errors" => {
+  #   "something" => {
+  #     "somethingElse" => {
+  #       "password" => ["VALIDATION_STATE_PASSWORD"]
+  #     }
+  #   }
+  # }
+  #
+  def nest
```

Comment:
Since this turns an Array into a Hash, perhaps calling this `to_nested_hash` or something similar would be more self-documenting? It'd also follow the whole `to_str`, `to_a`, `to_json`, naming convention then..